### ISSUE TITLE
Mongoose error : buffering timed out after 10000ms

### ISSUE THREAD

**Issue by AntoineIN** [Automattic/mongoose#9732]
Hello, i got this error on mongoose with nodejs :

version of mongoose : 5.11.8

I'm trying to save data on my mongodb database via mongoose. But i got this error everytime : 
`MongooseError: Operation `usermodels.insertOne()` buffering timed out after 10000ms`

I'm sure that my server is connected to the database.

This is my code : 

```javascript
const MongoClient = require('mongodb').MongoClient;
const mongoose = require('mongoose');

const uri = `link for my database`;
const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

client.connect()
    .then(() => {
        console.log('start');
    })
    .catch(err => {
        console.error('App starting error:', err.stack);
        process.exit(1)
    });

const Schema = mongoose.Schema;

const UserSchema = new Schema({
    user: String,
    usermail: String,
    userpwd: String,
    updated: { type: Date, default: Date.now },
});

const user_instance = mongoose.model('UserModel', UserSchema );

const myUser = new user_instance({ 
    user: "React JS",
    usermail: "test@test.com",
    userpwd: "teest",
});

myUser.save(function (err) { 
    console.log(err) 
});
```

Any idea ?
Thanks you !

**Comment by ParasiteTM** [comment]
Check if you have the correct network access to your cluster.

**Comment by kkhanhluu** [comment]
You are using mongo client to connect to database. Try to connect to mongodb database by using [mongoose.connect() or mongoose.createConnection()](https://mongoosejs.com/docs/connections.html)

**Comment by xuke-hat** [comment]
I faced the same issue. My mongoose's version is 5.11.8 and node's version is 14.15.1. 

I'm following a tutorial of using MongoDb Atlas. I runned the following code and the connection was fine. But when I tried to insert something it showed ```MongooseError: Operation `notes.insertOne()` buffering timed out after 10000ms``` 

```javascript
const url = `mongodb+srv://notes:xxx@cluster0.fwxrn.mongodb.net/note-app?retryWrites=true&w=majority`

const mongoose = require('mongoose')
mongoose.connect(url, { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false, useCreateIndex: true })

const noteSchema = new mongoose.Schema({
  content: String,
  date: Date,
  important: Boolean,
})

const Note = mongoose.model('Note', noteSchema)

const note = new Note({
  content: 'HTML is Easy',
  date: new Date(),
  important: true,
})

note.save().then(result => {
  console.log('note saved!')
  mongoose.connection.close()
})
```

**Comment by kkhanhluu** [comment]
@xuke-hat did you try to remove node_modules folder and reinstall _mongoose_ package? This [answer](https://stackoverflow.com/questions/65090440/how-to-solve-mongoose-v5-11-0-model-find-error-operation-products-find-bu) can be helpful

**Comment by xuke-hat** [comment]
@kkhanhluu The answer didn't work but thanks for your help.

Now I've solved the error. I thought the timeout in insertion implied that the connection was fine, but I'm wrong. It showed that I failed to connect to the database, though I checked my password many times and the IP whitelist is 0.0.0.0/0. Finally, I followed this [answer](https://stackoverflow.com/questions/61800062/error-querytxt-etimeout-when-connecting-to-mongodb-atlas-using-mongoose) and changed my DNS to 8.8.8.8 and it works.

**Comment by vkarpov15** [comment]
@AntoineIN your script isn't actually using Mongoose to connect. You're using the MongoDB driver's `MongoClient` class, but then you aren't actually telling Mongoose to connect to MongoDB. That's why you're getting this error.

Replace

```javascript
const client = new MongoClient(uri, { useNewUrlParser: true, useUnifiedTopology: true });

client.connect().then(...)
```

With:

```javascript
mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true }).then(...)
```

Please read the [connection docs](https://mongoosejs.com/docs/connections.html).

**Comment by Ding-Fan** [comment]
In my situation, I followed this answer https://stackoverflow.com/a/63912290/6236633 and made it working

**Comment by gugateider** [comment]
The problem I faced was that I was able to connect to my MongoDB Atlas through Compass but wasn't able to connect through my app. Had exactly the same issue that you've had, tried all the solutions above but none of them worked.

The problem was that apparently, my DNS was being cached and once I restarted my router everything started working again.
I'm not sure if your problem is the same as mine but make sure you have whitelisted your IP and try restarting your router
